File: ruby/matrix/matrix.rb
class Matrix
  attr_reader :rows, :columns

  def initialize(text)
    @rows = text.split("\n").map do |row|
      row.split.map(&:to_i)
    end

    @columns = (0...@rows[0].length).map do |index|
      @rows.map { |row| row.at(index) }
    end
  end
end

File: javascript/matrix/matrix.js
export class Matrix {
  constructor(text) {

    this._rows =
      text
      .split('\n')
      .map(row => row.split(' ').map(c => Number(c)));
  }

  get rows() { return this._rows; }

  get columns() {
    return this._columns = this._columns ||
        this._rows[0]
          .map((_, index) => this._rows.map(row => row[index]));
  }
}
